developer_docs/syntactic_stage.md

```diff
@@ -0,0 +1,139 @@
+# Syntactic Stage
+
+The syntactic stage is described by the file [pyccel.parser.syntactic](../pyccel/parser/syntactic.py)
+
+The syntactic stage serves 4 main purposes:
+1.  [**Navigation and AST Creation**](#Navigation-and-ast-creation) : Convert Python's [AST](https://docs.Python.org/3/library/ast.html) (abstract syntax tree) representation of the Python file to Pyccel's AST representation (objects of the classes in the folder [pyccel.ast](../pyccel/ast))
+2.  [**Errors**](#Errors) : Raise an error for any syntax used that is not yet supported by pyccel
+3.  [**Headers**](#Headers) : Convert header comments from strings to Pyccel's AST representation
+4.  [**Scoping**](#Scoping) : Collect the name of all variables in each scope (see [scope](scope.md) for more details) to ensure no name collisions can occur if pyccel generates Variable names
+
+## Navigation and AST Creation
+
+The entry point for the class `SyntaxParser` is the function `parse`.
+This function is called from the constructor to examine a FST (Full Syntax Tree) object.
+The FST is the output of Python's `ast.parse` function with comments inserted using the [`pyccel.parser.extend_tree.extend_tree`](../pyccel/parser/extend_tree.py).
+Python's ast module discards all comments, so the function [`pyccel.parser.extend_tree.extend_tree`](../pyccel/parser/extend_tree.py) is needed to read the file and reinsert these comments at the correct location.
```

Comment:
I find that pyccel.parser.extend_tree.extend_tree is redundant. Wouldn't be better to write : 

```suggestion
The FST is the output of Python's `ast.parse` function with comments inserted at the correct location using the [`pyccel.parser.extend_tree.extend_tree`](../pyccel/parser/extend_tree.py), since Python's ast module discards all comments.
```
instead of lines 15 and 16 ?

Reply:
I agree. Better: a098fb89 ?

Reply:
yes, much better ?